pkg/il/compiler/compiler.go

```diff
@@ -206,6 +206,9 @@ func (g *generator) generateFunction(f *expr.Function, depth int, mode nilMode,
 		g.generateIndex(f, depth, mode, valueJmpLabel)
 	case "OR":
 		g.generateOr(f, depth, mode, valueJmpLabel)
+	case "ip":
```

Comment:
I think there is a great opportunity to generalize this code here by updating the default case instead.
default:
  for arg := range f.Args {
     g.generate(arg, ...)
  }
  g.builder.Call(f.Name)

Reply:
I think you are probably right, but getting that all sorted out is a bit beyond what I want to tackle in this PR. I've added a TODO and intend on coming back to this ASAP.  OK?

Reply:
A TODO sounds good.